### PR TITLE
Convert proxy factory auto generate mode to integer

### PR DESCRIPTION
**Pull request by dragosprotung**

### CODE REVIEW

File: lib/Doctrine/Common/Proxy/Exception/InvalidArgumentException.php [hunk 1]
```diff
@@ -109,4 +109,14 @@ public static function classMustNotBeFinal($className)
     {
         return new self(sprintf('Unable to create a proxy for a final class "%s".', $className));
     }
+
+    /**
+     * @param mixed $value
+     *
+     * @return self
+     */
+    public static function invalidAutoGenerateMode($value)
```
Comment (lcobucci) [inline lib/Doctrine/Common/Proxy/Exception/InvalidArgumentException.php]: Please add the return type declation

File: tests/Doctrine/Tests/Common/Proxy/AbstractProxyFactoryTest.php [hunk 2]
```diff
@@ -14,6 +14,52 @@
...
+    {
+        $proxyGenerator = $this->createMock(ProxyGenerator::class);
+        $metadataFactory = $this->createMock(ClassMetadataFactory::class);
+
+        $proxyFactory = $this->getMockForAbstractClass(
+            AbstractProxyFactory::class,
+            [$proxyGenerator, $metadataFactory, $autoGenerate]
+        );
+
+        $this->assertAttributeSame($expected, 'autoGenerate', $proxyFactory);
```
Comment (lcobucci) [inline tests/Doctrine/Tests/Common/Proxy/AbstractProxyFactoryTest.php]: Please use `self::assert*()`

File: tests/Doctrine/Tests/Common/Proxy/AbstractProxyFactoryTest.php [hunk 3]
```diff
@@ -14,6 +14,52 @@
...
+        ];
+    }
+
+    /**
+     * @dataProvider dataAutoGenerateValues
+     *
+     * @param mixed $autoGenerate
+     * @param int $expected
+     */
+    public function testNoExceptionIsThrownForValidIntegerAutoGenerateValues($autoGenerate, $expected)
```
Comment (lcobucci) [inline tests/Doctrine/Tests/Common/Proxy/AbstractProxyFactoryTest.php]: Please add the type declaration on `$expected` and ` : void` as return type

**Comment by dragosprotung** [comment]
I changed the commit base to master and resolved in GitHub UI the conflict and this made a bit of a mess. Don't know how to have a cleaner PR. Shall I close this one and submit a new one ?

**Comment by lcobucci** [comment]
@dragosprotung we usually use `git rebase` to sync branches and solve conflicts. It's up to you to decide which path you want to go. Personally I'd advise to give it a try, you might learn about some nice git features :wink:

**Comment by lcobucci** [comment]
@dragosprotung we usually use `git rebase` to sync branches and solve conflicts. It's up to you to decide which path you want to go. Personally I'd advise to give it a try, you might learn about some nice git features :wink:

**Comment by dragosprotung** [comment]
Usually I use rebase, but not this time.
Submitted  #816

**Pull request closed without merging**